README.md

```diff
@@ -85,15 +85,18 @@ and the following Typescript types:
 - `GoResult<T> = { data: T; success: true }`
 - `GoResultSuccess<E extends Error = Error> = { error: E; success: false }`
 - `GoResultError<T, E extends Error = Error> = GoResultSuccess<T> | GoResultError<E>`
-- `PromiseOptions { readonly retries?: number; readonly retryDelayMs?: number; readonly timeoutMs?: number; }`
+- `GoAsyncOptions { readonly retries?: number; readonly timeoutMs?: number; }`
 
-The default values for `PromiseOptions` are:
+The default values for `GoAsyncOptions` are:
 
 ```ts
-{ retries: 0, retryDelay: 200, timeoutMs: 0 }
+// retries = number of retries to be made
+// timeoutMs = number of milliseconds to wait for all attempts
+{ retries: 0, timeoutMs: 0 }
```

Comment:
Note that I've removed the support for retry delay (see "tiny api" docs section below)

Reply:
I mean..that's all nice and well but we're not building a general-purpose library for users, we're building a library that we need for our projects. And we do need a retry delay (https://github.com/api3dao/airnode/blob/master/packages/airnode-utilities/src/evm/gas-prices/gas-prices.ts#L16) and we do need a retry with a random backoff delay (Airkeeper, Airseeker). So, it doesn't need to be a part of this PR but we need these.

Reply:
> I didn't like this delay property because it is not very flexible and forces the same delay between retries.

I've never said that it has to be done that way.

> I don't agree with this argument. It's true we are not building it for users, but for us. The appropriate question is whether this should be implemented in promise utils or in the individual projects. Promise utils is a very low level library (I'd say it's a design pattern) so it makes sense to keep the API surface minimal.
> At first I thought, we should implement these in the Airseeker and Airkeeper completely (and leave this package minimal), but on the second thought it makes sense to provide some "generic" support to do this via promise utils. My main point is to avoid bloated configuration (such as: https://github.com/lifeomic/attempt#usage).

I totally agree with not bloating the configuration, we don't need more than half of the options from attempt library. But I don't agree with keeping this minimalist/low level/design pattern library. In fact, you can argue we should create a library for everything from the `airnode-utils` package. But that's just not practical. We need libraries that help us do stuff. If I know that our use case is to wait some time between retries the library should support it. It makes no sense to keep it minimal and implement the rest where we need it. Especially when we need it in multiple places.

> For that reason I propose adding beforeAttempt(attemptIndex) and afterAttempt(attemptIndex, goRes) functions. Both can be async and both will be executed before/after a go callback is invoked. I think the logic doing the random/fixed retry should be implemented in the individual projects.

This makes no sense. We don't need any `beforeAttempt` and `afterAttempt` callbacks. We never needed the functionality like this. What we need (and know we need) is option to setup either a static delay between retries (not sure we even need this but is the simplest) and a random delay between retries. That's it. If the library doesn't provide that and I need to implement it every time I want to use it the library is half-useless. The point of libraries is to hide commonly used functionality so you don't need to implement it in the project yourself.

I'm totally fine with coming up with a clear and minimal configuration options but the functionality must be included and usable with no hustle. That is what you want from the library.

Let's make the configuration simple. Options:
* **retries** - number of retires (we already have that)
* **timeoutMs** - timeout for each attempt in ms (we already have that)
* **delayType** - either `static` or `random`, if not set any other delay options will throw an error
* **delayMs** - for static delay, delay in ms
* **minDelayMs** - for random delay, minimal delay in ms
* **maxDelayMs** - for random delay, maximal delay in ms

That's it, nothing more. we can even omit the `delayType` but I think it's cleaner with it. Usage is clear and it does exactly what we need.

Reply:
I get your point. I still disagree with parts of it but that stems from my opinion that I'd like this library to be more generic and minimal, which I agree we do not need.

I don't like the options you've suggested because it's not clear what happens when you mix e.g. delayType static + minDelayMs. I'd rather implement it like this:
- retries
- timeoutMs
- delay = object which will have type and other properties for that type.